Formula/matterhorn.rb

```diff
@@ -0,0 +1,20 @@
+class Matterhorn < Formula
+  desc "Unix terminal client for the Mattermost chat system"
+  homepage "https://github.com/matterhorn-chat/matterhorn"
+  url "https://github.com/matterhorn-chat/matterhorn.git", :tag => "50200.1.1"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+
+  def install
+    system "rm", "cabal.project"
+    system "cabal", "new-update"
+    system "cabal", "install", "tasty-quickcheck"
+    system "./build.sh"
+    bin.install "dist-newstyle/build/x86_64-osx/ghc-8.4.4/matterhorn-50200.1.1/x/matterhorn/build/matterhorn/matterhorn"
+  end
+
+  test do
+    system "#{bin}matterhorn", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
It won't be easy because `matterhorn` is a chat client running in a terminal. Starting it open the following user login and password interface:

```
┌──────────────────────────────────────────────────────────┐
│Please enter your Mattermost credentials to log in.       │
│                                                          │
│Server hostname:                                          │
│                                                          │
│Server port:      443                                     │
│                                                          │
│Username:                                                 │
│                                                          │
│Password:                                                 │
│                                                          │
│          Press Enter to log in or Esc to exit.           │
└──────────────────────────────────────────────────────────┘
```